core-primitives/api-client-extensions/src/chain.rs

```diff
@@ -33,6 +34,16 @@ where
 		self.get_signed_block(hash)
 	}
 
+	fn get_blocks(&self, from: u32, to: u32) -> ApiResult<Vec<SignedBlock>> {
+		let mut blocks = Vec::<SignedBlock>::new();
+
+		for n in from..=to {
+			let block = self.get_signed_block_by_num(Some(n)).unwrap().unwrap();
+			blocks.push(block);
+		}
```

Comment:
Please don't introduce any unwraps. The first one can be replaced by an `?`. The second one contains the `Option<SignedBlock>`. I would recommend the following:

```
if let Some(block) = self.get_signed_block_by_num(Some(n))? {
  blocks.push(block);
}
```

Reply:
I know we have `unwraps` in quite some places, but in many of them, they are in the core loop of the worker, where we actually want to abort if we get an error. Here, however, we have library code, which could be called from many places; hence we should not `unwrap` and let the caller decide what to with the error.

Reply:
I totally agree with @clangenb - just one more minor comment: In tests it is acceptable to use `unwrap()` for simplification, since there it makes sense to fail with a panic that is caught by the test framework.